source/tutorials/packaging-existing-software.md

```diff
@@ -446,21 +446,28 @@ In case all else fails, it helps to become familiar with searching the [Nixpkgs
 ### Git and `rg`
 
 To find name assignments in the source, search for `"<keyword> ="`.
-For example, these are the search results for [`"x11 = "`](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22x11+%3D%22&type=code) or [`"libx11 ="`](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22libx11+%3D%22&type=code) on Github .
+For example, these are the search results for [`"x11 = "`](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22x11+%3D%22&type=code) or [`"libx11 ="`](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22libx11+%3D%22&type=code) on Github.
 
-Or fetch a local clone of the repository and use `rg`.
-Nixpkgs is huge.
-Only clone the latest revision if you don't want to wait a long time:
+Or fetch a local clone of the repository and use `rg` (a nicer `grep` implementation, found in the `ripgrep` package).
 
+Since you might not have `git` or `rg` installed on your system, let's start a shell with these two tools:
 ```console
 $ nix-shell -p git ripgrep
+[nix-shell:~]$ # git & rg are available in this shell!
+```
+
+Since the Nixpkgs repository is huge, it can be faster to only clone the latest revision to avoid waiting a long time for a full clone:
+
+```console
 [nix-shell:~]$ git clone https://github.com/NixOS/nixpkgs --depth 1
+...
+[nix-shell:~]$ cd nixpkgs/
```

Comment:
I also noticed that all prompts below this line are actually wrong, since we did a `cd nixpkgs/` they should be like:
```diff
-[nix-shell:~]$ some commands....
+[nix-shell:~/nixpkgs]$ some commands...
```

Do you want me to change these as well?